src/runner.py

```diff
@@ -640,9 +622,9 @@ def _configure_docker_proxy(self):
         docker_client_proxy = {
             "proxies": {
                 "default": {
-                    "httpProxy": self.config.proxies["http"],
-                    "httpsProxy": self.config.proxies["https"],
-                    "noProxy": self.config.proxies["no_proxy"],
+                    "httpProxy": self.config.proxies.http or "",
+                    "httpsProxy": self.config.proxies.https or "",
+                    "noProxy": self.config.proxies.no_proxy or "",
```

Comment:
It would probably better not to set the environment variables rather than setting them to empty strings if the proxy information is not defined. You can do this concisely using something like `{key: value for key, value in (("httpProxy", self.config.proxies.http), ...) if value}`